pages/collective-page.tsx

```diff
@@ -53,27 +54,48 @@ const getPropsFromContext = (context: Context) => {
 
 const collectivePageQueryHelper = getSSRQueryHelpers<
   ReturnType<typeof getVariablesFromContext>,
-  ReturnType<typeof getPropsFromContext>
+  ReturnType<typeof getPropsFromContext>,
+  // TODO: Use the query auto-generated type when we move this to V2
+  { Collective: { slug: string; id: number; name: string; isHost: boolean; host?: { slug: string } } }
 >({
   query: collectivePageQuery,
   getVariablesFromContext,
   getPropsFromContext,
-  preload: (client, result) => preloadCollectivePageGraphqlQueries(client, result?.data?.Collective),
+  preload: (client, result) => preloadCollectivePageGraphqlQueries(client, result?.Collective),
 });
 
 // next.js export
 // ts-unused-exports:disable-next-line
-export const getServerSideProps: typeof collectivePageQueryHelper.getServerSideProps = async (context: Context) => {
+export const getServerSideProps = async (context: Context) => {
+  const whitelabel = getWhitelabelProps(context);
   const result = await collectivePageQueryHelper.getServerSideProps(context);
   const props = result['props'];
-  const notFound = collectivePageQueryHelper.checkResultContainsNonNullResult(props, 'Collective(');
+  const error = props[APOLLO_ERROR_PROP_NAME];
+  const data = props[APOLLO_QUERY_DATA_PROP_NAME];
+  const notFound = !error && collectivePageQueryHelper.checkResultContainsNonNullResult(props, 'Collective({"slug"');
+
   if (context.res && context.req) {
     const { locale } = getRequestIntl(context.req);
     if (locale === 'en') {
       context.res.setHeader('Cache-Control', 'public, s-maxage=300');
     }
   }
 
+  if (!whitelabel?.isNonPlatformDomain && data?.Collective) {
+    const whitelabelRedirect = shouldRedirect(
+      data?.Collective?.host?.slug || data?.Collective?.slug,
+      data?.Collective?.slug,
+    );
+    if (whitelabelRedirect) {
+      return {
+        redirect: {
+          destination: `${whitelabelRedirect.domain}/${whitelabelRedirect.slug}`,
+          permanent: false,
+        },
+      };
+    }
+  }
+
```

Comment:
In the future, I would prefer an implementation where the GraphQL return a property like collective.canonicalDomain or collective.canonicalUrl. The API should be responsible for that, not the frontend.

If moving with the prototype in production, I want to be able to tweak the canonical domain just by editing a setting in the Collective settings or data field.